### PR TITLE
DROOLS-2593 : [Guided Decision Table] Trouble to add a row into table with WI

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-2593

### CODE REVIEW

**Comment by Rikkola** [comment]
@manstis @jomarko If you could please review.
@barboras7 Ping, in case you are interested too.

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-api/src/main/java/org/drools/workbench/services/verifier/api/client/index/WorkItemAction.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.services.verifier.api.client.index;
+
+import org.drools.workbench.services.verifier.api.client.configuration.AnalyzerConfiguration;
+import org.drools.workbench.services.verifier.api.client.index.keys.Values;
+
+public class WorkItemAction
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-api/src/main/java/org/drools/workbench/services/verifier/api/client/index/WorkItemAction.java]: @Rikkola please update formatting, I some not necessary white spaces nearby constructor brackets.

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-backend/src/test/java/org/drools/workbench/services/verifier/webworker/client/fromfile/BRLFragmentsAnalyzerFromFileTest.java [hunk 2]
```diff
@@ -59,4 +60,27 @@ public void testRuleTableGDST() throws
...
+
+        analyzer.resetChecks();
+        analyzer.analyze();
+
+        final Set<Issue> analysisReport = analyzerProvider.getAnalysisReport();
+
+        for (final Issue issue : analysisReport) {
+            System.out.println(issue.getCheckType());
+            System.out.println(issue.getDebugMessage());
+        }
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-backend/src/test/java/org/drools/workbench/services/verifier/webworker/client/fromfile/BRLFragmentsAnalyzerFromFileTest.java]: I think we should remove this for cycle or use some logger.

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-client/src/main/java/org/drools/workbench/services/verifier/plugin/client/api/HeaderMetaData.java [hunk 3]
```diff
@@ -30,7 +31,23 @@ public HeaderMetaData(@MapsTo("patternsByColumnNumber") final Map<Integer, Model
         this.patternsByColumnNumber = patternsByColumnNumber;
     }
 
-    public Map<Integer, ModelMetaData> getPatternsByColumnNumber() {
-        return patternsByColumnNumber;
+    public ModelMetaData getPatternsByColumnNumber(Integer columnIndex) {
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-client/src/main/java/org/drools/workbench/services/verifier/plugin/client/api/HeaderMetaData.java]: Could we use `final Integer columnIndex` please?

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-client/src/main/java/org/drools/workbench/services/verifier/plugin/client/builders/PatternResolver.java [hunk 4]
```diff
@@ -88,17 +88,15 @@ public Pattern resolve() {
 
     private String getFactType() {
         final String factType = headerMetaData
-                .getPatternsByColumnNumber()
-                .get(PortablePreconditions.checkNotNull("columnIndex",
+                .getPatternsByColumnNumber(PortablePreconditions.checkNotNull("columnIndex",
                                                         columnIndex))
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-client/src/main/java/org/drools/workbench/services/verifier/plugin/client/builders/PatternResolver.java]: I think also here kiegroup formatting needs to be applied.

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-client/src/main/java/org/drools/workbench/services/verifier/plugin/client/builders/PatternResolver.java [hunk 5]
```diff
@@ -88,17 +88,15 @@ public Pattern resolve() {
...
                 .getFactType();
         return factType;
     }
 
     private String getBoundName() {
         final String boundName = headerMetaData
-                .getPatternsByColumnNumber()
-                .get(PortablePreconditions.checkNotNull("columnIndex",
+                .getPatternsByColumnNumber(PortablePreconditions.checkNotNull("columnIndex",
                                                         columnIndex))
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-client/src/main/java/org/drools/workbench/services/verifier/plugin/client/builders/PatternResolver.java]: I think also here kiegroup formatting needs to be applied.

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-client/src/main/java/org/drools/workbench/services/verifier/plugin/client/api/HeaderMetaData.java [hunk 6]
```diff
@@ -30,7 +31,23 @@ public HeaderMetaData(@MapsTo("patternsByColumnNumber") final Map<Integer, Model
...
+        if (patternsByColumnNumber.containsKey(columnIndex)) {
+            return patternsByColumnNumber.get(columnIndex);
+        } else {
+            throw new IllegalArgumentException("Could not find columnIndex: " + columnIndex);
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-client/src/main/java/org/drools/workbench/services/verifier/plugin/client/api/HeaderMetaData.java]: Could we cover also this case in `ModelMetaDataEnhancerTest` please?

**Comment by Rikkola** [comment]
@jomarko Updated the PR

**Comment by jomarko** [comment]
jenkins execute full downstream build

Because local build log this error to console log:
```
ERROR [org.drools.workbench.services.verifier.plugin.backend.VerifierWebWorkerServlet] (default task-126) Failed to load verifier web worker.
```

**Comment by Rikkola** [comment]
The web worker need to be built with "mvn -Dfull clean install" for it to show up. Idea is not smart enough to pick it up and compile. I guess it could be with the right start parameters.

**Comment by Rikkola** [comment]
@jomarko No problem. This is really different than the usual way GWT works.

**Comment by jomarko** [comment]
jenkins please retest this

**Comment by barboras7** [comment]
jenkins please retest this

**Comment by Rikkola** [comment]
There was some drools-pmml dependency issues.

jenkins execute full downstream build

**Comment by Rikkola** [comment]
jenkins execute full downstream build

**Comment by Rikkola** [comment]
jenkins execute full downstream build

**Comment by Rikkola** [comment]
@manstis Hi, this can now be merged. Thank you.

**Pull request merged**